mutation_partition.cc

```diff
@@ -1962,19 +1962,22 @@ void query_result_builder::consume_end_of_stream() {
 }
 
 stop_iteration query::result_memory_accounter::check_local_limit() const {
-    if (_total_used_memory > _maximum_result_size.hard_limit) {
-        if (_short_read_allowed) {
+    if (_short_read_allowed) {
+        if (_total_used_memory > _maximum_result_size.get_page_size()) {
```

Comment:
Suggestion: we can just return `stop_iteration(_total_used_memory > _maximum_result_size.get_page_size())` here and then we can deindent the part below.

Reply:
Good idea. Will do.